### ISSUE TITLE
ValueError: Protocol message AutoprovisioningNodePoolDefaults has no "management" field.

### ISSUE THREAD

**Issue by rjribeiro** [googleapis/python-container#118]
I have this code running inside google cloud composer:
```
from google.cloud.container_v1.types import (
     AutoprovisioningNodePoolDefaults,
     NodeManagement
)

management = NodeManagement(auto_repair=False)
     node_pool_nap = AutoprovisioningNodePoolDefaults(oauth_scopes=["https://www.googleapis.com/auth/cloud-platform"],
                                                      management=management)

```

Composer is returning the error message:
```
node_pool_nap = AutoprovisioningNodePoolDefaults(oauth_scopes=["https://www.googleapis.com/auth/cloud-platform"],
ValueError: Protocol message AutoprovisioningNodePoolDefaults has no "management" field.
```

In [documentation](https://googleapis.dev/python/container/latest/container_v1/types.html#google.cloud.container_v1.types.AutoprovisioningNodePoolDefaults) and here in [repo](https://github.com/googleapis/python-container/blob/master/google/cloud/container_v1/types/cluster_service.py) shows that this field exists

#### Environment details

composer-1.17.0-preview.2-airflow-2.0.1

google-cloud-container==1.0.1

#### Steps to reproduce

  1. Load the code below in the bucket of gcp composer version composer-1.17.0-preview.2-airflow-2.0.1

#### Code example

```python
from airflow import DAG
from airflow.operators.dummy import DummyOperator
from airflow.utils.dates import days_ago
from google.cloud.container_v1.types import (
     AutoprovisioningNodePoolDefaults,
     NodeManagement
)

management = NodeManagement(auto_repair=False)
node_pool_nap = AutoprovisioningNodePoolDefaults(oauth_scopes=["https://www.googleapis.com/auth/cloud-platform"],
                                                      management=management)
args = {
    'owner': 'airflow',
}

dag = DAG(dag_id="test", default_args=args, start_date=days_ago(2))
DummyOperator(task_id="dummy", dag=dag)

**Comment by askmeegs** [comment]
Hi Rafael, sorry for the delay here. I was able to reproduce this bug -- 
<img width="1564" alt="reproduced-bug" src="https://user-images.githubusercontent.com/3137106/148283075-9d83b832-ac95-4b58-93e0-e1326734ad93.png">

Using the same code + setup you have, except for composer version you used is no longer available, so I'm on `composer-1.17.7-airflow-2.0.2`. 

<img width="624" alt="composer-available-versions" src="https://user-images.githubusercontent.com/3137106/148283130-dfdd81a2-2b83-4117-a1be-09bae46ef3a1.png">

Then when I edited my DAG to print out the attributes of a blank `AutoprovisioningNodePoolDefaults` object, `management` doesn't appear, along with others that should, given the [documentation here](https://googleapis.dev/python/container/latest/container_v1/types.html#google.cloud.container_v1.types.AutoprovisioningNodePoolDefaults).  Here's what appears in Airflow logs when I run the `dir()` command on one of these objects:


```
[2022-01-05 20:10:03,017] {gke.py:37} INFO - ['ByteSize', 'Clear', 'ClearExtension', 'ClearField', 'CopyFrom', 'DESCRIPTOR', 'DiscardUnknownFields', 'Extensions', 'FindInitializationErrors', 'FromString', 'HasExtension', 'HasField', 'IsInitialized', 'ListFields', 'MergeFrom', 'MergeFromString', 'ParseFromString', 'RegisterExtension', 'SerializePartialToString', 'SerializeToString', 'SetInParent', 'UnknownFields', 'WhichOneof', '_CheckCalledFromGeneratedFile', '_SetListener', '__class__', '__deepcopy__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getstate__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__setstate__', '__sizeof__', '__slots__', '__str__', '__subclasshook__', '__unicode__', '_extensions_by_name', '_extensions_by_number', 'oauth_scopes', 'service_account']

```

I'm going to reach out to the Cloud Composer team to determine exactly what version of the Python GKE client library is running under the hood, and will update this issue after. thanks!

**Comment by askmeegs** [comment]
Hello! Following up on this, the Apache Airflow team responded to that issue ^ that they are deliberately limiting google-cloud-container versions to < 2.0 due to some breaking changes that were introduced. ([docs here](https://googleapis.dev/python/container/latest/UPGRADING.html))

https://github.com/apache/airflow/pull/10317 

Given this, and the hypothesis that this bug is caused by the out of date version of google-cloud-container missing certain object fields like `management` closing this issue as infeasible - until at which date airflow (and in turn, Cloud Composer) upgrades to beyond 2.0.0.
